src/Paket.Core/DependenciesFile.fs

```diff
@@ -398,6 +398,27 @@ type DependenciesFile(fileName,groups:Map<GroupName,DependenciesGroup>, textRepr
 
     member __.Groups = groups
 
+    member this.SimplifyFrameworkRestrictions() = 
+        let transform (dependenciesFile:DependenciesFile) (group:DependenciesGroup) =
+            if group.Options.Settings.FrameworkRestrictions <> [] then dependenciesFile else
+            match group.Packages with
+            | [] -> dependenciesFile
+            | package::rest ->
+                let sameRequirements =
+                    rest |> Seq.forall (fun p' -> p'.Settings.FrameworkRestrictions = package.Settings.FrameworkRestrictions)
+
+                if not sameRequirements || package.Settings.FrameworkRestrictions = [] then dependenciesFile else
```

Comment:
must check if restrictions are empty? does it harm if you add empty restrictions globally?